File: internal/domain/bookmark.go
package domain

import "time"

type Bookmark struct {
	ID          string    `json:"id"`
	UserID      string    `json:"user_id"`
	URL         string    `json:"url"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	ImageURL    string    `json:"image_url"`
	Category    string    `json:"category"`
	Tags        []string  `json:"tags"`
	Enriched    bool      `json:"enriched"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateBookmarkInput struct {
	URL         string
	Title       string
	Description string
	ImageURL    string
	Category    string
	Tags        []string
}

File: ios/Boxmind/Core/Models/Bookmark.swift
import Foundation

struct Bookmark: Codable, Identifiable, Equatable, Hashable, Sendable {
    let id: String
    let userID: String
    let url: String
    let title: String
    let description: String
    let imageURL: String
    let category: String
    let tags: [String]
    let folderID: String?
    let enriched: Bool
    let createdAt: Date
    let updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case userID = "user_id"
        case url
        case title
        case description
        case imageURL = "image_url"
        case category
        case tags
        case folderID = "folder_id"
        case enriched
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct CreateBookmarkRequest: Encodable, Sendable {
    let url: String
}
